lib/dependencies/HarmonyDetectionParserPlugin.js

```diff
@@ -69,9 +53,7 @@ module.exports = class HarmonyDetectionParserPlugin {
 				);
 			}
 			if (!parser.state.harmonyModule) {
-				throw new Error(
-					"Top-level-await is only supported in EcmaScript Modules"
-				);
+				this.checkHarmony(parser, true);
```

Comment:
Sadly this won't work as `parser.state.harmonyModule` need to be set before parsing the AST as it changes behavior of some elements. Technically we could walk the AST in the `program` hook, but this would double parse the Source so it's probably not worth it.

So I'll cancel this card and users have to opt into ESM by some other method. Probably `"type": "module"` in package.json will become the goto method anyway so this is not so important.